### ISSUE TITLE
Allow migrations with resources including a space

### ISSUE THREAD

**Issue by entropitor** [minamijoyo/tfmigrate#5]
It seems tfmigrate splits on a space for a `mv` command and only allows 1 space after `mv `, however, this basically means we cannot write `state mv` commands for resources in a for_each with a key that includes a space

**Comment by minamijoyo** [comment]
@entropitor Thank you for opening issue!

I didn't know the fact that an address of resource could contain spaces.
To clarify what are you trying, could you provide a sample Terraform configuration and state mv command?

**Comment by entropitor** [comment]
```terraform
resource "docker_image" "nginx" {
  name  = "nginx:latest"
  keep_locally = false
}

resource "docker_container" "nginx" {
  for_each = {
    "This is an example" = "example"
  }

  image = docker_image.nginx.latest
  name = each.value
}
```

Output of `terraform state list`
```
docker_container.nginx["This is an example"]
docker_image.nginx
```

Output of `tf state show 'docker_container.nginx["This is an example"]'` (I truncated it a bit to hide some potential sensitive values)
```
# docker_container.nginx["This is an example"]:
resource "docker_container" "nginx" {
    attach            = false
    command           = [
        "nginx",
        "-g",
        "daemon off;",
    ]
    cpu_shares        = 0
    dns               = []
    dns_opts          = []
    entrypoint        = [
        "/docker-entrypoint.sh",
    ]
    image             = "sha256:f35646e83998b844c3f067e5a2cff84cdf0967627031aeda3042d78996b68d35"
    name              = "example"
    network_mode      = "default"
    privileged        = false
    publish_all_ports = false
    read_only         = false
    restart           = "no"
    rm                = false
    shm_size          = 64
    start             = true
}
```

**Comment by minamijoyo** [comment]
Wow 😮 it contains space and also double-quote...

The current implementation of parsing action string is here:

https://github.com/minamijoyo/tfmigrate/blob/7b868807be43ccf3505dd0d46ab1071917f25273/tfmigrate/state_action.go#L27
https://github.com/minamijoyo/tfmigrate/blob/7b868807be43ccf3505dd0d46ab1071917f25273/tfmigrate/multi_state_action.go#L26

We probably need to escape special characters. `go-shellwords` may help us to simulate shell escape logic.

**Comment by entropitor** [comment]
@minamijoyo Does it actually make sense that you do need to do parsing? I don't know hcl very well but wouldn't it make more sense if the developer already provided the splitted string?

**Comment by minamijoyo** [comment]
@entropitor Using such a schema-less string for action is intentional by design. It allows us to easily copy terraform state command to action.

Every design choice has pros and cons. I know the strictly typed-schema you mentioned will provide us fine-grained controls. Actually my initial idea was it. However, I wrote some examples of migration file and found that it's very redundant to write migration files. If I want to split a tfstate and move 10 resources to other tfstate, it requires approximately 4 lines x 10 resources = 40 lines. In addition, it also added cognitive load to users. Most of Terraform users know the terraform state command syntax, but probably don't remember what is a correct name of attribute for each action in migration file.

I prefer to a design solving most common use-cases simply rather than imposing complexity on users only to support a minor edge case.

**Comment by entropitor** [comment]
Okay, then I guess your solution seems like the best way to go forward

**Comment by minamijoyo** [comment]
@entropitor Fixed in v0.1.1.

Note that we need to escape double-quote ( `\"` ). It's a HCL limitation.

```hcl
migration "state" "test" {
  dir = "tmp/test5"
  actions = [
    "mv docker_container.nginx 'docker_container.nginx[\"This is an example\"]'",
  ]
}
```

**Comment by entropitor** [comment]
Nice thanks! That's one of the things I tried before so makes sense
